rebasebot/bot.py

```diff
@@ -333,19 +334,24 @@ def _is_pr_available(dest_repo, rebase: GitHubBranch):
     except StopIteration:
         pass
 
+    logging.info("No existing pull request")
     return "", False
 
 
 def _create_pr(
         gh_app: github3.GitHub,
         dest: GitHubBranch,
         source: GitHubBranch,
-        rebase: GitHubBranch
+        rebase: GitHubBranch,
+        workdir: str
 ):
+    gitwd = git.Repo.init(path=workdir)
+    source_head_commit = gitwd.git.rev_parse(f"source/{source.branch}")
+
     logging.info("Creating a pull request")
 
     pull_request = gh_app.repository(dest.ns, dest.name).create_pull(
-        title=f"Merge {source.url}:{source.branch} into {dest.branch}",
+        title=f"Merge {source.url}:{source.branch} ({source_head_commit}) into {dest.branch}",
```

Comment:
Is this the full commit sha or can we limit this substitution to the normal 7 char short sha?

Reply:
Good point, done.
Title result can be seen here: https://github.com/openshift/cluster-api-provider-aws/pull/456

Reply:
Will the PR title ever be updated if the source commit changes? Will this cause the bot to open many PRs rather than reusing the same one? Is there any logic based on the PR title being constant?

Reply:
That's a good point.
It didn't use to update the title.
Now it updates the title correctly with: https://github.com/openshift-eng/rebasebot/pull/30/commits/44fdbfeb611598da88ae2bf67b6b96f8099dae2f

See
<img width="1034" alt="retitled" src="https://user-images.githubusercontent.com/4902157/217007752-71d49692-5511-46bc-98bc-f9813ae590aa.png">